public/static/docs/api-reference/open.md

```diff
@@ -0,0 +1,188 @@
+> details, please refer to Python's
+> [`open()`](https://docs.python.org/3/library/functions.html#open) built-in,
+> which is used under the hood.
+
+`dvc.api.open()` may only be used as a
+[context manager](https://www.python.org/dev/peps/pep-0343/#context-managers-in-the-standard-library)
+(using the `with` keyword, as shown in the **Examples** below).
+
+> Use `dvc.api.read()` to get the file's contents directly – no _context
+> manager_ involved.
+
+This function reads (streams) the file trough a direct connection to the storage
+whenever possible, so it does not require any space on the disc to save the file
+before making it accessible. The only exception is when using a Google Drive
+[remote type](/doc/command-reference/remote/add#supported-storage-types).
+
+## Parameters
+
+- **`path`** - location and file name of the file in `repo`, relative to the
+  project's root.
+
+- `repo` - specifies the location of the DVC project. It can be a URL or a file
+  system path. Both HTTP and SSH protocols are supported for online Git repos
+  (e.g. `[user@]server:project.git`). _Default_: The current project is used
+  (the current working directory tree is walked up to find it).
+
+- `rev` - Git commit (any [revision](https://git-scm.com/docs/revisions) such as
+  a branch or tag name, or a commit hash). If `repo` is not a Git repo, this
+  option is ignored. _Default_: `HEAD`.
+
+- `remote` - name of the [DVC remote](/doc/command-reference/remote) to look for
+  the target data. _Default_: The
+  [default remote](/doc/command-reference/remote/default) of `repo` is used if a
+  `remote` argument is not given. For local projects, the <abbr>cache</abbr> is
+  tied before the default remote.
+
+- `mode` - specifies the mode in which the file is opened. Defaults to `"r"`
+  (read). Mirrors the namesake parameter in builtin
+  [`open()`](https://docs.python.org/3/library/functions.html#open).
+
+- `encoding` -
+  [codec](https://docs.python.org/3/library/codecs.html#standard-encodings) used
+  to decode the file contents to a string. This should only be used in text
+  mode. Defaults to `"utf-8"`. Mirrors the namesake parameter in builtin
+  `open()`.
+
+## Exceptions
+
+- `dvc.exceptions.FileMissingError` - file in `path` is missing from `repo`.
+
+- `dvc.exceptions.PathMissingError` - `path` cannot be found in `repo`.
+
+- `dvc.api.UrlNotDvcRepoError` - `repo` is not a DVC project.
+
+- `dvc.exceptions.NoRemoteError` - no `remote` is found.
+
+## Example: Use data tracked in a DVC repository online
+
+Any <abbr>data artifact</abbr> can be employed directly in your Python app by
+using this API.
+
+For example, an XML file from a public DVC repo online can be processed directly
+in your Python app with:
+
+```py
+from xml.dom.minidom import parse
+import dvc.api
+
+with dvc.api.open(
+        'get-started/data.xml',
+        repo='https://github.com/iterative/dataset-registry'
+        ) as fd:
+    xmldom = parse(fd)
```

Comment:
super relevant example would to show SAX or StAX parser instead of a DOM one - that's where it shines. Or we can make CSV example the main one and show how we process it in steam fashion (e.g. calculating sum or avg) - it would show the "streaming" aspect of the `open()` way better.

Reply:
> make CSV example the main one and show how we process it in steam fashion (e.g. calculating sum or avg

Thinking about this, I don't think we're talking about real-time data streaming so that continuously calculating a metric would be logical. Or maybe I missed the point?

Reply:
I merged this PR but still thinking about this one. What's the advantage of streaming files in open/read? Probably just making a big file available quickly so you can start processing it before it's all downloaded, but again, I don't think you'll want to show the progress of such processing, or is that a major use case you guys see? Cc @Suor @shcheklein

Reply:
Moved this discussion to a new PR: #1037